apps/dashboard/app/models/remote_transfer.rb

```diff
@@ -0,0 +1,157 @@
+class RemoteTransfer < Transfer
+
+  validates_each :src_remote, :dest_remote do |record, _, remote|
+    remote_type = RcloneUtil.remote_type(remote)
+    if remote_type.nil? && remote != 'fs'
+      record.errors.add :base, "Remote #{remote} does not exist"
+    elsif ::Configuration.allowlist_paths.present? && (remote_type == 'local' || remote_type == 'alias')
+      record.errors.add :base, "Remotes of type #{remote_type} are not allowed due to ALLOWLIST_PATH"
+    end
+  end
+
+  validates_each :files do |record, _, files|
+    files.each do |k, v|
+      # Validate paths in the same was as PosixTransfer for the local filesystem (fs)
+      if record.src_remote == "fs"
+        record.errors.add :base, "#{k} is not included under ALLOWLIST_PATH" unless AllowlistPolicy.default.permitted?(k.to_s)
+      end
+      if record.dest_remote == "fs"
+        # rm commands are [{ k => nil}] - nil values
+        record.errors.add :base, "#{v} is not included under ALLOWLIST_PATH" if !v.nil? && !AllowlistPolicy.default.permitted?(v.to_s)
+      end
+    end
+
+    if record.action == 'mv' || record.action == 'cp'
+      # local filesystem
+      if record.dest_remote == "fs"
+        conflicts = files.values.select { |f| File.exist?(f) }
+        record.errors.add :base, "These files already exist: #{conflicts.join(', ')}" if conflicts.present?
+      else
+        # remote
+        begin
+          existing = RcloneUtil.lsf(record.dest_remote, record.to).map { |file| File.join(record.to, file) }
+          conflicts = files.values.intersection(existing)
+          record.errors.add :base, "These files already exist: #{conflicts.join(', ')}" if conflicts.present?
+        rescue RcloneError => e
+          if e.exitstatus != 3
+            # Rclone will return status 3 if directory doesn't exist (ok), other errors are unexpected
+            record.errors.add :base, "Error checking existing files in destination: #{e}"
+          end
+        end
+      end
+    end
+  end
+
+  attr_accessor :src_remote, :dest_remote, :filesizes, :transferred
+
+  class << self
+    def transfers
+      # all transfers stored in the Transfer class
+      Transfer.transfers
+    end
+
+    def build(action:, files:, src_remote:, dest_remote:)
+      if files.is_a?(Array)
+        # rm action will want to provide an array of files
+        # so if it is an Array we convert it to a hash:
+        #
+        # convert [a1, a2, a3] to {a1 => nil, a2 => nil, a3 => nil}
+        files = Hash[files.map { |f| [f, nil] }]
+      end
+
+      self.new(action: action, files: files, src_remote: src_remote, dest_remote: dest_remote)
+    end
+  end
+
+  # total number of bytes
+  def steps
+    return @steps if @steps
+
+    @filesizes = {}
```

Comment:
```suggestion
    @filesizes = {}.with_indifferent_access
```

I see we're using symbols all over the place, but `with_indifferent_access` just makes it easy to forget/not care.